### PR TITLE
added in AppContext and audio player

### LINKED ISSUE #1: Audio player element is missing from Phonograph component
I made this issue on my **phone** while walking to get coffee. Sick!

### PR DESCRIPTION
**Pull request by franknoirot**
Resolves #1, #2, and #4.

## Additional Work

- added nav links (no rollover yet)
-got the audio player playing continuously while navigating by making the nav a global component.
- added animations to the music notes while the song is playing.

### CODE REVIEW

**Comment by robpetrin** [comment]
Hey @franknoirot eager to look into where this failed a little more. Is the netlify org for neta centrally-controlled or with you? I think payment begins after you add your second person, so it's not needed to add me if the creds are Frank-specific. But I can't look into this more without that. :D Thanks!

**Comment by franknoirot** [comment]
Hey @robpetrin sorry the checks that were failing were left over from the Sanity starter that I had the site set up with originally. The starter created two sites in Netlify, the frontend and the CMS, both of which got built when the repo was pushed to. Though I had removed the Sanity Studio code from the repo, the site was still configured to try to rebuild, which of course failed. I deleted that site out of Netlify and invited you to the organization in there.

**Comment by robpetrin** [comment]
Oh my goodness this really might change everything huh

File: components/Navbar/Navbar.js [hunk 1]
```diff
@@ -1,19 +1,39 @@
...
+        return router.pathname === url
+    }
+
     return (
         <header className={styles.navBar + ' double-drop'}>
             <div className={styles.topNav}>
                 <Phonograph />
-                <Image className={styles.monogram} width="160" height="160" src='/./img/monogram.svg' alt="L + G"/>
+                <Link href='/'>
+                    <a>
```
Comment (franknoirot) [inline components/Navbar/Navbar.js:19]: Interesting bit of syntax, the NextJS Link component requires an `<a>` tag within it to work.

File: components/Phonograph/Phonograph.module.css [hunk 2]
```diff
@@ -44,6 +44,19 @@
     fill: none;
 }
 
-.active .notes {
-    fill: #DBE2C5;
+.active .notes path {
+    animation: play-notes 3s steps(2, end) infinite;
+}
+
+.active  .notes path:nth-of-type(2) {
```
Comment (franknoirot) [inline components/Phonograph/Phonograph.module.css:51]: The notes all animate the same, just with a slight offset.

File: lib/AppContext.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            {children}
+        </AppContext.Provider>
+    )
+}
+
+export function useAppContext() {
+    return useContext(AppContext)
+}
+
+function appReducer(state, action) {
```
Comment (franknoirot) [inline lib/AppContext.js:24]: Read this [bible](https://reactjs.org/docs/hooks-reference.html) for more info on useReducer and useContext. Way overkill for this, but very useful for larger projects.

**Review by franknoirot — COMMENTED** [review]
Just left some explainer comments.

File: components/Navbar/Navbar.js (hunk 1 above)
Comment (robpetrin) [reply components/Navbar/Navbar.js:19]: interesting -- vanilla React does not, right?

**Review by robpetrin — APPROVED** [review]
Hi @franknoirot -- looks great! Excited to collaborate more on these, and good call on the review requirements for PRs to main. No red flags here -- merging. See #5 for a future idea for our prettier 💅  -- thanks!

**Pull request merged**